test/Microsoft.AspNet.Http.Tests/DefaultHttpContextTests.cs

```diff
@@ -154,68 +154,63 @@ public void SetItems_NewCollectionUsed()
-            context.UpdateFeatures(newFeatures);
+            newFeatures.Set<IHttpRequestFeature>(new HttpRequestFeature());
+            newFeatures.Set<IHttpResponseFeature>(new HttpResponseFeature());
+            newFeatures.Set<IHttpWebSocketFeature>(new TestHttpWebSocketFeature());
+            context.Initialize(newFeatures);
 
-            Assert.Equal(0, newFeatures.Count());
+            Assert.Equal(3, newFeatures.Count());
 
-            TestCachedFeaturesAreNull(context.GetType(), context, newFeatures);
-            TestCachedFeaturesAreNull(request.GetType(), request, newFeatures);
-            TestCachedFeaturesAreNull(response.GetType(), response, newFeatures);
-            TestCachedFeaturesAreNull(authentication.GetType(), authentication, newFeatures);
-            TestCachedFeaturesAreNull(connection.GetType(), connection, newFeatures);
-            TestCachedFeaturesAreNull(websockets.GetType(), websockets, newFeatures);
+            TestCachedFeaturesAreNull(context, newFeatures);
+            TestCachedFeaturesAreNull(context.Request, newFeatures);
+            TestCachedFeaturesAreNull(context.Response, newFeatures);
+            TestCachedFeaturesAreNull(context.Authentication, newFeatures);
+            TestCachedFeaturesAreNull(context.Connection, newFeatures);
+            TestCachedFeaturesAreNull(context.WebSockets, newFeatures);
 
             context.Session = new TestSession();
-            newFeatures.Set<IHttpRequestFeature>(new HttpRequestFeature());
-            newFeatures.Set<IHttpResponseFeature>(new HttpResponseFeature());
-            newFeatures.Set<IHttpWebSocketFeature>(new TestHttpWebSocketFeature());
 
-            TestCachedFeaturesAreSet(context.GetType(), context, newFeatures);
-            TestCachedFeaturesAreSet(request.GetType(), request, newFeatures);
-            TestCachedFeaturesAreSet(response.GetType(), response, newFeatures);
-            TestCachedFeaturesAreSet(authentication.GetType(), authentication, newFeatures);
-            TestCachedFeaturesAreSet(connection.GetType(), connection, newFeatures);
-            TestCachedFeaturesAreSet(websockets.GetType(), websockets, newFeatures);
+            TestCachedFeaturesAreSet(context, newFeatures);
+            TestCachedFeaturesAreSet(context.Request, newFeatures);
+            TestCachedFeaturesAreSet(context.Response, newFeatures);
+            TestCachedFeaturesAreSet(context.Authentication, newFeatures);
+            TestCachedFeaturesAreSet(context.Connection, newFeatures);
+            TestCachedFeaturesAreSet(context.WebSockets, newFeatures);
 
-            Assert.NotEqual(0, newFeatures.Count());
+            Assert.NotEqual(3, newFeatures.Count());
         }
 
-        void TestCachedFeaturesAreNull(Type type, object value, IFeatureCollection features)
+        void TestCachedFeaturesAreNull(object value, IFeatureCollection features)
         {
+            var type = value.GetType();
 
             var fields = type
```

Comment:
This doesn't verify anything anymore because the object layout has changed. The only field it will find is `FeatureReferences<FeatureInterfaces> _features;`.